dashbuilder/dashbuilder-backend/dashbuilder-services/src/main/java/org/dashbuilder/transfer/DataTransferServicesImpl.java

```diff
@@ -0,0 +1,364 @@
+        org.uberfire.backend.vfs.Path pluginPath =
+           org.uberfire.backend.server.util.Paths.convert(
+               Paths.get(uri).resolve(newFilePath));
+
+        Plugin plugin = new Plugin(
+            newFile.toPath().getParent().getFileName().toString(),
+            TypeConverterUtil.fromPath(pluginPath),
+            pluginPath);
+
+        pluginAddedEvent.fire(new PluginAdded(plugin, sessionInfo));
+    }
+
+    private void fireNavigationEvent() {
+        navTreeChangedEvent.fire(
+            new NavTreeChangedEvent(
+                navTreeStorage.loadNavTree()));
+    }
+
+    private FileSystem getImportFileSystem(File file, String tempPath) {
+        List<FileSystem> fileSystems = Arrays.asList(datasetsFS, perspectivesFS, navigationFS);
+        String filePath = file.toURI().toString();
+
+        return fileSystems.stream()
+            .filter(fs -> filePath.contains(tempPath + fs.getName()))
+            .findFirst()
+            .orElse(null);
+    }
+
+    private void moveZipToFileSystem(String zipLocation, FileSystem fileSystem) {
+        String sourceLocation = new StringBuilder()
+            .append(SpacesAPI.Scheme.FILE)
+            .append("://")
+            .append(zipLocation)
+            .toString();
+
+        Path source = Paths.get(URI.create(sourceLocation));
+
+        Path target = Paths.get(
+        	URI.create(
+        		new StringBuilder()
+        			.append(SpacesAPI.Scheme.GIT)
+		        	.append("://")
+		        	.append(fileSystem.getName())
+		        	.append(File.separator)
+		        	.append(FILE_PATH)
+		        	.append(File.separator)
+		            .append(EXPORT_FILE_NAME)
+		            .toString()));
+
+        ioService.write(target, Files.readAllBytes(source));
+
+        Files.delete(source);
+    }
+
+    private File unzipFile(File destinationDir, ZipEntry zipEntry, ZipInputStream zis) throws Exception {
+        File destFile = new File(destinationDir, zipEntry.getName());
+
+        if (!destFile.exists()) {
+            destFile.getParentFile().mkdirs();
+            destFile.createNewFile();
+        }
+
+        FileOutputStream fos = new FileOutputStream(destFile);
+        int len;
+        while ((len = zis.read(buffer)) > 0) {
+            fos.write(buffer, 0, len);
+        }
+        fos.close();
+
+        return destFile;
+    }
+
+    private void zipFileSystem(FileSystem fs, ZipOutputStream zos) {
+        Path root = fs.getRootDirectories().iterator().next();
+        Files.walkFileTree(root, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path path, BasicFileAttributes attrs) throws IOException {
+                try {
+                    String location = fs.getName() + path.toString();
+                    zipFile(path.toFile(), location, zos);
+                    return FileVisitResult.CONTINUE;
+
+                } catch (Exception e) {
+                    e.printStackTrace();
```

Comment:
I suggest logging this exception instead of printing the stacktrace

Reply:
done